### PR TITLE
Add a `from_ase_atoms()` method to `Structure`

### PR DESCRIPTION
**Pull request by Andrew-S-Rosen**
## Summary

Adds a `.from_ase_atoms()` method to `Structure` to complement the pre-existing `.to_ase_atoms()`. In reality, this is just a shorthand for the adapter, but it felt awkward to have one direction but not the other.

## Checklist

- [X] Google format doc strings added. Check with `ruff`.
- [X] Type annotations included. Check with `mypy`.
- [X] Tests added for new features/fixes.
- [X] If applicable, new classes/functions/modules have [`duecredit`](https://github.com/duecredit/duecredit) `@due.dcite` decorators to reference relevant papers by DOI ([example](https://github.com/materialsproject/pymatgen/blob/91dbe6ee9ed01d781a9388bf147648e20c6d58e0/pymatgen/core/lattice.py#L1168-L1172))

Tip: Install `pre-commit` hooks to auto-check types and linting before every commit:

```sh
pip install -U pre-commit
pre-commit install
```

## Summary by CodeRabbit

- **New Features**
	- Introduced a method for converting `ase.Atoms` to `pymatgen Structure`, enhancing interoperability between different scientific libraries.
- **Tests**
	- Added new assertions to ensure reliability of the new conversion functionality.

### CODE REVIEW

File: tests/core/test_structure.py [hunk 1]
```diff
@@ -1848,9 +1790,10 @@ def test_to_ase_atoms(self):
         assert isinstance(atoms, Atoms)
         assert len(atoms) == len(self.struct)
         assert AseAtomsAdaptor.get_structure(atoms) == self.struct
+        assert Structure.from_ase_atoms(atoms) == self.struct
```
[the remark this answers is not in the document]
Comment (Andrew-S-Rosen) [reply tests/core/test_structure.py:1851]: Those are checked implicitly via the equality operator, thank you very much.

File: pymatgen/core/structure.py [hunk 2]
```diff
@@ -939,6 +905,20 @@ def to_ase_atoms(self, **kwargs) -> Atoms:
...
+            kwargs: Passed to AseAtomsAdaptor.get_structure.
+        
+        Returns:
+            Structure
+        """
+
+        from pymatgen.io.ase import AseAtomsAdaptor
+
+        return AseAtomsAdaptor.get_structure(self, **kwargs)
+
```
[the remark this answers is not in the document]
Comment (Andrew-S-Rosen) [reply pymatgen/core/structure.py]: No.

**Review by janosh — APPROVED** [review]
nice one, thanks @Andrew-S-Rosen! 👍

**Comment by JaGeo** [comment]
Now I can simplify a lot of my lecture code. 😅
Thanks, @Andrew-S-Rosen !!

**Comment by Andrew-S-Rosen** [comment]
@shyuep --- just chiming in with another vote to remove coderabbit. It's super annoying. If not though, I'll just keep sass talking back to it until it spontaneously decides not to review anymore.

**Comment by shyuep** [comment]
@Andrew-S-Rosen Yes, it is just an experiment. But a question - are the proposed changes useful? I can understand the boiler is annoying and I can try to kill those off. Just wanted to know if there is any value or there is no value at all.

**Comment by Andrew-S-Rosen** [comment]
> @Andrew-S-Rosen Yes, it is just an experiment. But a question - are the proposed changes useful? I can understand the boiler is annoying and I can try to kill those off. Just wanted to know if there is any value or there is no value at all.

In general, [not really](https://github.com/materialsproject/pymatgen/pull/3812#discussion_r1591753624). I'm sure sometimes the proposed changes are useful, but the signal:noise ratio (both in terms of the chattiness and usefulness of the suggestions) is likely too low for contributors to get much value from it in my opinion. The only redeeming feature, in my view, is the edit of the original summary.

**Comment by shyuep** [comment]
@Andrew-S-Rosen I don't really care for the summary. Ok, I will disable it. 

One thing - I am not sure I want a from_ase_atoms. I would prefer a simple Structure.from_object. this will support ASE atoms for now, but I am sure there are a lot of different formats we can convert a structure from.

**Pull request merged**